app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php

```diff
@@ -0,0 +1,109 @@
+
+class PersistentTest extends TestCase
+{
+    /**
+     * @var Persistent
+     */
+    private $customerData;
+
+    /**
+     * @var Session|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $persistentSessionHelperMock;
+
+    /**
+     * @var View|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerViewHelperMock;
+
+    /**
+     * @var CustomerRepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerRepositoryMock;
+
+    /**
+     * Setup environment for test
+     */
+    protected function setUp()
+    {
+        $this->persistentSessionHelperMock = $this->createMock(Session::class);
+        $this->customerViewHelperMock = $this->createMock(View::class);
+        $this->customerRepositoryMock = $this->createMock(CustomerRepositoryInterface::class);
+
+        $objectManager = new ObjectManagerHelper($this);
+
+        $this->customerData = $objectManager->getObject(
+            Persistent::class,
+            [
+                'persistentSession' => $this->persistentSessionHelperMock,
+                'customerViewHelper' => $this->customerViewHelperMock,
+                'customerRepository' => $this->customerRepositoryMock
+            ]
+        );
+    }
+
+    /**
+     * Test getSectionData() when disable persistent
+     */
+    public function testGetSectionDataWhenDisablePersistent()
+    {
+        $this->persistentSessionHelperMock->method('isPersistent')->willReturn(false);
+
+        $this->assertEquals([], $this->customerData->getSectionData());
+    }
+
+    /**
+     * Test getSectionData() when customer doesn't login
+     */
+    public function testGetSectionDataWithNotLogin()
+    {
+        $this->persistentSessionHelperMock->method('isPersistent')->willReturn(true);
+
+        $persistentSessionMock = $this->createPartialMock(PersistentSession::class, ['getCustomerId']);
+        $persistentSessionMock->method('getCustomerId')->willReturn(null);
+        $this->persistentSessionHelperMock->method('getSession')->willReturn($persistentSessionMock);
+
+        $this->assertEquals([], $this->customerData->getSectionData());
+    }
+
+    /**
+     * Test getSectionData() when customer login and enable persistent
+     */
+    public function testGetSectionDataCustomerLoginAndEnablePersistent()
+    {
+        $this->persistentSessionHelperMock->method('isPersistent')->willReturn(true);
+
+        $persistentSessionMock = $this->createPartialMock(PersistentSession::class, ['getCustomerId']);
+        $persistentSessionMock->method('getCustomerId')->willReturn(1);
+        $this->persistentSessionHelperMock->method('getSession')->willReturn($persistentSessionMock);
+
+        $customerMock = $this->createMock(CustomerInterface::class);
+        $this->customerRepositoryMock->method('getById')->with(1)->willReturn($customerMock);
+        $this->customerViewHelperMock->method('getCustomerName')->with($customerMock)->willReturn('Adam John');
```

Comment:
`Adam John` is another example of magic literal. You should replace that with STUB_CUSTOMER_NAME